### PR TITLE
Compactify the implementation of ContourLabeler.add_label_near.

### PR DESCRIPTION
**Pull request by anntzer**
The old implementation was quite expansive; compactify it a bit.  Also
reformat the docs.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/matplotlib/contour.py [hunk 1]
```diff
@@ -455,64 +455,47 @@ def add_label_clabeltext(self, x, y, rotation, lev, cvalue):
...
-
         if transform:
             x, y = transform.transform((x, y))
 
         # find the nearest contour _in screen units_
         conmin, segmin, imin, xmin, ymin = self.find_nearest_contour(
             x, y, self.labelIndiceList)[:5]
 
-        # The calc_label_rot_and_inline routine requires that (xmin, ymin)
+        # calc_label_rot_and_inline requires that (xmin, ymin)
```
Comment (timhoffm) [inline lib/matplotlib/contour.py:486]:
```suggestion
        # calc_label_rot_and_inline() requires that (xmin, ymin)
```
Indicates that it's a function (call).

**Review by timhoffm — APPROVED** [review]

File: lib/matplotlib/contour.py [hunk 2]
```diff
@@ -455,64 +455,47 @@ def add_label_clabeltext(self, x, y, rotation, lev, cvalue):
...
         xcmin = self.axes.transData.inverted().transform([xmin, ymin])
-        # if there isn't a vertex close enough
-        if not np.allclose(xcmin, lc[imin]):
-            # insert new data into the vertex list
-            lc = np.row_stack([lc[:imin], xcmin, lc[imin:]])
-            # replace the path with the new one
-            paths[segmin] = mpath.Path(lc)
+        if not np.allclose(xcmin, lc[imin]):  # If no vertex is close enough...
+            lc = np.insert(lc, imin, xcmin, axis=0)  # add new pt in vertices,
+            paths[segmin] = mpath.Path(lc)  # and replace path by new one.
```
Comment (QuLogic) [inline lib/matplotlib/contour.py:494]: Don't like abbreviating comments just to save a line/fit in the line length
```suggestion
        if not np.allclose(xcmin, lc[imin]):
            # If no vertex is close enough, add new point in vertices, and
            # replace path by new one.
            lc = np.insert(lc, imin, xcmin, axis=0)
            paths[segmin] = mpath.Path(lc)
```

File: lib/matplotlib/contour.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/contour.py]: yup

File: lib/matplotlib/contour.py (hunk 2 above)
Comment (anntzer) [reply lib/matplotlib/contour.py]: yup

**Review by QuLogic — APPROVED** [review]

**Pull request merged**